rules_poetry/poetry.bzl

```diff
@@ -21,12 +30,18 @@ def _get_python_interpreter_attr(rctx):
 def _resolve_python_interpreter(rctx):
     python_interpreter = _get_python_interpreter_attr(rctx)
 
-    if rctx.attr.python_interpreter_target:
+    if "win" in rctx.os.name:
+        interpreter_target = rctx.attr.python_interpreter_target_win
+    elif "mac" in rctx.os.name:
+        interpreter_target = rctx.attr.python_interpreter_target_mac
+    else:
+        interpreter_target = rctx.attr.python_interpreter_target
+    
+    if interpreter_target:
         if rctx.attr.python_interpreter:
-            fail("python_interpreter_target and python_interpreter incompatible")
-
-        target = rctx.attr.python_interpreter_target
-        python_interpreter = rctx.path(target)
+            fail("interpreter_target and python_interpreter incompatible")
```

Comment:
Can you comment what's going on here? It's not intuitive why rctx.attr.python_interpreter being defined is problematic.

Reply:
add comments in [7aa2a09](https://github.com/mongodb-forks/rules_poetry/pull/1/commits/7aa2a09847cc9482b937f4381021a9cb45946821)